cmd/abapEnvironmentPullGitRepo.go

```diff
@@ -75,34 +72,36 @@ func runAbapEnvironmentPullGitRepo(options *abapEnvironmentPullGitRepoOptions, t
 	client.SetOptions(clientOptions)
 	pollIntervall := com.GetPollIntervall()
 
-	log.Entry().Infof("Start pulling %v repositories", len(options.RepositoryNames))
-	for _, repositoryName := range options.RepositoryNames {
-
-		log.Entry().Info("-------------------------")
-		log.Entry().Info("Start pulling " + repositoryName)
-		log.Entry().Info("-------------------------")
-
-		// Triggering the Pull of the repository into the ABAP Environment system
-		uriConnectionDetails, errorTriggerPull := triggerPull(repositoryName, connectionDetails, client)
-		if errorTriggerPull != nil {
-			return errors.Wrapf(errorTriggerPull, "Pull of '%s' failed on the ABAP System", repositoryName)
+	repositories := []abaputils.Repository{}
+	err = checkPullRepositoryConfiguration(*options)
 
-		}
+	if err == nil {
+		repositories, err = abaputils.GetRepositories(&abaputils.RepositoriesConfig{RepositoryNames: options.RepositoryNames, Repositories: options.Repositories})
+	}
 
-		// Polling the status of the repository import on the ABAP Environment system
-		status, errorPollEntity := abaputils.PollEntity(repositoryName, uriConnectionDetails, client, pollIntervall)
-		if errorPollEntity != nil {
-			return errors.Wrapf(errorPollEntity, "Pull of '%s' failed on the ABAP System", repositoryName)
-		}
-		if status == "E" {
-			return errors.New("Pull of " + repositoryName + " failed on the ABAP System")
-		}
+	if err == nil {
+		err = pullRepositories(repositories, connectionDetails, client, pollIntervall)
+	}
 
-		log.Entry().Info(repositoryName + " was pulled successfully")
+	if err != nil {
+		return fmt.Errorf("Something failed during the pull of the repositories: %w", err)
```

Comment:
Same as above. Wrapping the error with "Something failed during the pull of the repositories:" does not provide valuable info (and would be misleading, if the error occurred in abaputils.GetRepositories)

Reply:
No, the error does provide valuable info. Exactly this case even is tested within Unit tests, e.g. in file `abapEnvironmentCheckoutBranch_test.go` in line 170 with expected error message in line 126 stating: `Something failed during the checkout: Could not parse config file repositoriesTest.yml, AddonDescriptor doesn't contain any repositories`

Taking error wrapping into account in my eyes this provides all valuable information needed.
If you check for error messages in abaputils.GetRepositories you will exactly see what failed. 

So what would you change in this particular case? I don't see any useful changes that could be added. Also there will be an error stack trace that further reduces this issue IMO.